examples/synchronous-proxy/workflows.rb

```diff
@@ -0,0 +1,122 @@
+require_relative "proxy/communications"
+require_relative "activities"
+
+module SynchronousProxy
+  RegisterStage = "register".freeze
+  SizeStage = "size".freeze
+  ColorStage = "color".freeze
+  ShippingStage = "shipping".freeze
+
+  TShirtSizes = ["small", "medium", "large"]
+  TShirtColors = ["red", "blue", "black"]
+
+  OrderStatus = Struct.new(:order_id, :stage, keyword_init: true)
+  TShirtOrder = Struct.new(:email, :size, :color) do
+    def to_s
+      "size: #{size}, color: #{color}"
+    end
+  end
+
+  class OrderWorkflow < Temporal::Workflow
+    include Proxy::Communications # defines #receive_request, #receive_response, #send_error_response, #send_request, and #send_response
+
+    timeouts start_to_close: 60
+
+    def execute
+      order = TShirtOrder.new
+      setup_signal_handler
+
+      # Loop until we receive a valid email
+      loop do
+        signal_detail = receive_request("email_payload")
+        source_id, email = signal_detail.calling_workflow_id, signal_detail.value
+        value, err = RegisterEmailActivity.execute(email)
```

Comment:
Should we use future for failure handling here? The current pattern looks like it was borrowed from Golang :)

Reply:
I was intentionally trying to mimic the golang version of the code so it was easy to translate the concepts. I usually prefer idiomatic Ruby. Can change this if you feel strongly about it. Then I may do further refactoring too since these loops are duplicative too.

Reply:
I think we should stick to idiomatic Ruby as much as possible. Especially in the examples section that will probably be used quite a lot by people implementing their own workflows/activities

Reply:
Replaced all of the `value, err = SomeCall` constructs with futures.